### DISCUSSION TITLE
Encountered invalid refName: "=="

### DISCUSSION THREAD

**Discussion by AlexWeinreb** [GMOD/jbrowse-components#2237] [Q&A] [answered]
I installed an instance of jbrowse2 on a server, then uploaded a fasta with its fai and gzi. When I open the page on a browser and select "Linear Genome View", I get a "Fatal Error" popup with this message:

```
Error: Encountered invalid refName: "�="
    in u
    in u
    in Ay
    in g
    in C
    in t
    in Unknown
    in Suspense
```
(the specific wrong refName might be slightly different, I had trouble copy-pasting)

This error message was for jbrowse v1.0.4, in v1.3.3 the same error occurs but only with the first line (no "in xxx").

The genome has standard ASCII sequence names:

```
$ zcat 210819_data/references/c_elegans.PRJNA13758.WS277.genomic_bgzip.fa.gz | grep ">"
>I
>II
>III
>IV
>V
>X
>MtDNA

$ cat 210819_data/references/c_elegans.PRJNA13758.WS277.genomic_bgzip.fa.gz.fai
I       15072434        3       60      61
II      15279421        15323649        60      61
III     13783801        30857733        60      61
IV      17493829        44871269        60      61
V       20924180        62656665        60      61
X       17718942        83929585        60      61
MtDNA   13794   101943850       60      61

$ cat config.json
{
  "assemblies": [
    {
      "name": "c_elegans.PRJNA13758.WS277.genomic_bgzip",
      "sequence": {
        "type": "ReferenceSequenceTrack",
        "trackId": "c_elegans.PRJNA13758.WS277.genomic_bgzip-ReferenceSequenceTrack",
        "adapter": {
          "type": "BgzipFastaAdapter",
          "fastaLocation": {
            "uri": "c_elegans.PRJNA13758.WS277.genomic_bgzip.fa.gz"
          },
          "faiLocation": {
            "uri": "c_elegans.PRJNA13758.WS277.genomic_bgzip.fa.gz.fai"
          },
          "gziLocation": {
            "uri": "c_elegans.PRJNA13758.WS277.genomic_bgzip.fa.gz.gzi"
          }
        }
      }
    }
  ],
  "configuration": {},
  "connections": [],
  "defaultSession": {
    "name": "New Session"
  },
  "tracks": []
}
```

And the problematic file can be generated with:
```
wget ftp://ftp.wormbase.org/pub/wormbase/releases/WS277/species/c_elegans/PRJNA13758/c_elegans.PRJNA13758.WS277.genomic.fa.gz
gunzip c_elegans.PRJNA13758.WS277.genomic.fa.gz
mv c_elegans.PRJNA13758.WS277.genomic.fa c_elegans.PRJNA13758.WS277.genomic_bgzip.fa
bgzip c_elegans.PRJNA13758.WS277.genomic_bgzip.fa
samtools faidx c_elegans.PRJNA13758.WS277.genomic_bgzip.fa.gz
# on server
jbrowse add-assembly 210819_data/references/c_elegans.PRJNA13758.WS277.genomic_bgzip.fa.gz --load symlink
```

At this point I don't know where the problem could come from, I'd welcome any pointer.

**Comment by cmdcolin** [comment, marked as answer]
hmm that all looks right! following those same steps locally worked for me. that is quite weird. thanks for giving the full details, it helps a lot though

you could possibly try giving the filename something different than the extension .gz

this is just a guess because this has in the past sometimes happened with vcf.gz files for example, the server incorrectly sets the "Content-Encoding: gzip" header and that causes the browser to unzip the gzip instead of jbrowse code unzipping the bgzip how it wants to

https://jbrowse.org/docs/faq_troubleshooting.html#what-is-the-error-invalid-bgzf-header-on-my-vcf-files

I have wondered if there is something jbrowse could do to help with this issue on the client side but generally it's not good when the server sends these headers

you could perhaps try hand-editing the config.json to make the fastaLocation uri not have a .gz extension c_elegans.PRJNA13758.WS277.genomic_bgzip.fa.anythingelsehere 

and then rename the file on disk appropriately

**Reply by AlexWeinreb** [reply]
I didn't manage to get the renaming to work, but simply using an uncompressed fasta seems to solve it, and is good enough for now. Thank you!

**Reply by rbuels** [reply]
This is a common enough problem that it might make sense for our code that uses bgzipped files to detect the presence of that header and skip decompression.
